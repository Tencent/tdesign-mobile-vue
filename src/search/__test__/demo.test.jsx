/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import actionVue from '@/search/demos/action.vue';
import baseVue from '@/search/demos/base.vue';
import maxLengthVue from '@/search/demos/max-length.vue';
import mobileVue from '@/search/demos/mobile.vue';
import otherVue from '@/search/demos/other.vue';
import shapeVue from '@/search/demos/shape.vue';

const mapper = {
  actionVue,
  baseVue,
  maxLengthVue,
  mobileVue,
  otherVue,
  shapeVue,
};

describe('Search', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Search ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
