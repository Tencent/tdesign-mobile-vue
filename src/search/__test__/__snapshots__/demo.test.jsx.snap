// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search > Search actionVue demo works fine 1`] = `
<div
  class="example-search"
>
  <div
    class="t-search"
  >
    <div
      class="t-search__input-box t-search__input-box--square"
    >
      <svg
        class="t-icon t-icon-search"
        fill="none"
        height="1em"
        style="font-size: 24px;"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M15.1 5.9a6.5 6.5 0 10-9.2 9.2 6.5 6.5 0 009.2-9.2zM4.49 4.5a8.5 8.5 0 0112.69 11.27l5.34 5.35-1.41 1.41-5.35-5.34A8.5 8.5 0 014.5 4.49z"
          fill="currentColor"
        />
      </svg>
      <input
        autofocus=""
        class="t-input__keyword t-search--center"
        placeholder="搜索预设文案"
        type="search"
      />
      <!---->
    </div>
    <!---->
  </div>
</div>
`;

exports[`Search > Search baseVue demo works fine 1`] = `
<div
  class="example-search"
>
  <div
    class="t-search"
  >
    <div
      class="t-search__input-box t-search__input-box--square"
    >
      <svg
        class="t-icon t-icon-search"
        fill="none"
        height="1em"
        style="font-size: 24px;"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M15.1 5.9a6.5 6.5 0 10-9.2 9.2 6.5 6.5 0 009.2-9.2zM4.49 4.5a8.5 8.5 0 0112.69 11.27l5.34 5.35-1.41 1.41-5.35-5.34A8.5 8.5 0 014.5 4.49z"
          fill="currentColor"
        />
      </svg>
      <input
        class="t-input__keyword"
        placeholder="请输入关键字"
        type="search"
      />
      <!---->
    </div>
    <!---->
  </div>
</div>
`;

exports[`Search > Search mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Search 搜索框123
  </h1>
  <p
    class="summary"
  >
    用于用户输入搜索信息，并进行页面内容搜索。
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        基础搜索框
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="example-search"
      >
        <div
          class="t-search"
        >
          <div
            class="t-search__input-box t-search__input-box--square"
          >
            <svg
              class="t-icon t-icon-search"
              fill="none"
              height="1em"
              style="font-size: 24px;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M15.1 5.9a6.5 6.5 0 10-9.2 9.2 6.5 6.5 0 009.2-9.2zM4.49 4.5a8.5 8.5 0 0112.69 11.27l5.34 5.35-1.41 1.41-5.35-5.34A8.5 8.5 0 014.5 4.49z"
                fill="currentColor"
              />
            </svg>
            <input
              class="t-input__keyword"
              placeholder="请输入关键字"
              type="search"
            />
            <!---->
          </div>
          <!---->
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        输入值后显示取消按钮
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="example-search"
      >
        <div
          class="t-search"
        >
          <div
            class="t-search__input-box t-search__input-box--square"
          >
            <svg
              class="t-icon t-icon-search"
              fill="none"
              height="1em"
              style="font-size: 24px;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M15.1 5.9a6.5 6.5 0 10-9.2 9.2 6.5 6.5 0 009.2-9.2zM4.49 4.5a8.5 8.5 0 0112.69 11.27l5.34 5.35-1.41 1.41-5.35-5.34A8.5 8.5 0 014.5 4.49z"
                fill="currentColor"
              />
            </svg>
            <input
              autofocus=""
              class="t-input__keyword t-search--center"
              placeholder="搜索预设文案"
              type="search"
            />
            <!---->
          </div>
          <!---->
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        02 组件样式
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        搜索框形状
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="example-search"
      >
        <div
          class="t-search"
        >
          <div
            class="t-search__input-box t-search__input-box--round"
          >
            <svg
              class="t-icon t-icon-search"
              fill="none"
              height="1em"
              style="font-size: 24px;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M15.1 5.9a6.5 6.5 0 10-9.2 9.2 6.5 6.5 0 009.2-9.2zM4.49 4.5a8.5 8.5 0 0112.69 11.27l5.34 5.35-1.41 1.41-5.35-5.34A8.5 8.5 0 014.5 4.49z"
                fill="currentColor"
              />
            </svg>
            <input
              class="t-input__keyword"
              placeholder="请输入关键字"
              type="search"
            />
            <!---->
          </div>
          <!---->
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        03 组件状态
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        默认状态其他对齐方式
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="example-search"
      >
        <div
          class="t-search"
        >
          <div
            class="t-search__input-box t-search__input-box--square"
          >
            <svg
              class="t-icon t-icon-search"
              fill="none"
              height="1em"
              style="font-size: 24px;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M15.1 5.9a6.5 6.5 0 10-9.2 9.2 6.5 6.5 0 009.2-9.2zM4.49 4.5a8.5 8.5 0 0112.69 11.27l5.34 5.35-1.41 1.41-5.35-5.34A8.5 8.5 0 014.5 4.49z"
                fill="currentColor"
              />
            </svg>
            <input
              class="t-input__keyword"
              placeholder="搜索预设文案"
              type="search"
            />
            <!---->
          </div>
          <!---->
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Search > Search otherVue demo works fine 1`] = `
<div
  class="example-search"
>
  <div
    class="t-search"
  >
    <div
      class="t-search__input-box t-search__input-box--square"
    >
      <svg
        class="t-icon t-icon-search"
        fill="none"
        height="1em"
        style="font-size: 24px;"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M15.1 5.9a6.5 6.5 0 10-9.2 9.2 6.5 6.5 0 009.2-9.2zM4.49 4.5a8.5 8.5 0 0112.69 11.27l5.34 5.35-1.41 1.41-5.35-5.34A8.5 8.5 0 014.5 4.49z"
          fill="currentColor"
        />
      </svg>
      <input
        class="t-input__keyword"
        placeholder="搜索预设文案"
        type="search"
      />
      <!---->
    </div>
    <!---->
  </div>
</div>
`;

exports[`Search > Search shapeVue demo works fine 1`] = `
<div
  class="example-search"
>
  <div
    class="t-search"
  >
    <div
      class="t-search__input-box t-search__input-box--round"
    >
      <svg
        class="t-icon t-icon-search"
        fill="none"
        height="1em"
        style="font-size: 24px;"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M15.1 5.9a6.5 6.5 0 10-9.2 9.2 6.5 6.5 0 009.2-9.2zM4.49 4.5a8.5 8.5 0 0112.69 11.27l5.34 5.35-1.41 1.41-5.35-5.34A8.5 8.5 0 014.5 4.49z"
          fill="currentColor"
        />
      </svg>
      <input
        class="t-input__keyword"
        placeholder="请输入关键字"
        type="search"
      />
      <!---->
    </div>
    <!---->
  </div>
</div>
`;
